src/AzureAppConfigurationImpl.ts

```diff
@@ -176,34 +176,70 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         };
     }
 
-    async #loadSelectedKeyValues(): Promise<ConfigurationSetting[]> {
-        const loadedSettings: ConfigurationSetting[] = [];
+    async #executeWithFailoverPolicy(funcToExecute) {
+        const clientWrappers = await this.#clientManager.getClients();
+        if (clientWrappers.length === 0) {
+            this.#clientManager.refreshClients();
+            throw new Error("No client is available to connect to the target App Configuration store.");
+        }
 
-        // validate selectors
-        const selectors = getValidKeyValueSelectors(this.#options?.selectors);
+        for (const clientWrapper of clientWrappers) {
+            let successful = false;
+            try {
+                const result = await funcToExecute(clientWrapper.client);
+                this.#isFailoverRequest = false;
+                successful = true;
+                updateClientBackoffStatus(clientWrapper, successful);
+                return result;
+            } catch (error) {
+                if (isFailoverableError(error)) {
+                    updateClientBackoffStatus(clientWrapper, successful);
+                    this.#isFailoverRequest = true;
+                    continue;
+                }
 
-        for (const selector of selectors) {
-            const listOptions: ListConfigurationSettingsOptions = {
-                keyFilter: selector.keyFilter,
-                labelFilter: selector.labelFilter
-            };
+                throw error;
+            }
+        }
 
-            const settings = listConfigurationSettingsWithTrace(
-                this.#requestTraceOptions,
-                this.#client,
-                listOptions
-            );
+        this.#clientManager.refreshClients();
+        throw new Error("All app configuration clients failed to get settings.");
+    }
 
-            for await (const setting of settings) {
-                if (!isFeatureFlag(setting)) { // exclude feature flags
-                    loadedSettings.push(setting);
+    async #loadSelectedKeyValues(): Promise<ConfigurationSetting[]> {
+        // validate selectors
+        const selectors = getValidKeyValueSelectors(this.#options?.selectors);
+        let loadedSettings: ConfigurationSetting[] = [];
+
+        const funcToExecute = async (client) => {
+            const loadedSettings: ConfigurationSetting[] = [];
+            for (const selector of selectors) {
+                const listOptions: ListConfigurationSettingsOptions = {
+                    keyFilter: selector.keyFilter,
+                    labelFilter: selector.labelFilter
+                };
+
+                const settings = listConfigurationSettingsWithTrace(
+                    this.#requestTraceOptions,
+                    client,
+                    listOptions
+                );
+
+                for await (const setting of settings) {
+                    if (!isFeatureFlag(setting)) { // exclude feature flags

```

Comment:
Here is what I mentioned before that feature flags are loaded twice. The settings of feature flag have been loaded but they will be discarded. And you will call `loadFeatureFlag` again in the `load` method. @Eskibear 

But this is another issue which unrelated with this PR.